### PR TITLE
Add docker-compose to spin up cardanon-node

### LINKED ISSUE #90: Use docker-compose for starting 3 Hydra nodes and one common Cardano node
Plan is to mainly reuse code from hydra-demo.

We also need to make sure that keys our CLI use are matching one used by Hydra nodes so this task is related to #87 .

### PR DESCRIPTION
**Pull request by nazrhom**
Closes #90

### CODE REVIEW

**Comment by GeorgeFlerovsky** [comment]
@uhbif19 Should we close this for now?

**Comment by nazrhom** [comment]
@GeorgeFlerovsky why close it? I though we had agreed to go with compose

**Comment by nazrhom** [comment]
I added the hydra-nodes and a script that starts everything and seeds the actors running the nodes with some ada and some fuel

**Comment by GeorgeFlerovsky** [comment]
Ah, OK. Nevermind. I got confused 😅

I'll go ahead and review it then 👍

File: stop-demo.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh -e
+
+docker-compose down
+docker-compose --profile hydra-node down
```
Comment (uhbif19) [inline stop-demo.sh:4]: Why we need profiles here?

Did not know about them BTW.

File: docker-compose.yaml [hunk 2]
```diff
@@ -2,19 +2,138 @@ version: "3.9"
 
 services:
   cardano-node:
-    image: inputoutput/cardano-node:1.35.4
+    image: inputoutput/cardano-node:latest
```
Comment (uhbif19) [inline docker-compose.yaml:5]: I am not sure if use latest is good, cuz `cardano-node` might break, while I guess we want need to change version often.

Comment (uhbif19) [reply docker-compose.yaml:5]:
```suggestion
    image: inputoutput/cardano-node:1.35.4
```

File: data/credentials/generate-actor.sh [hunk 3]
```diff
@@ -0,0 +1,3 @@
+cardano-cli address key-gen \
```
Comment (uhbif19) [inline data/credentials/generate-actor.sh:1]: I do not think that this is `data`.

File: data/hydra-keys/oscar.sk [hunk 4]
```diff
@@ -0,0 +1,5 @@
+{
```
Comment (uhbif19) [inline data/hydra-keys/oscar.sk:1]: I think that situation with those actors should be covered in docs somethere.

**Review by uhbif19 — CHANGES_REQUESTED** [review]

File: stop-demo.sh (hunk 1 above)
Comment (nazrhom) [reply stop-demo.sh:4]: We use the profile to start the hydra-nodes after the cardano-node (even though they are all in the same dockerfile)

File: seed-devnet.sh [hunk 5]
```diff
@@ -0,0 +1,108 @@
+#!/usr/bin/env bash
```
[the remark this answers is not in the document]
Comment (nazrhom) [reply seed-devnet.sh:1]: Yes, this step needs to happen to give the actors running the hydra nodes enough funds to manage the hydra lifecycle

File: docker-compose.yaml [hunk 6]
```diff
@@ -2,19 +2,138 @@ version: "3.9"
...
-      ]
\ No newline at end of file
+      , "--shelley-kes-key", "/devnet/kes.skey"
+      , "--shelley-vrf-key", "/devnet/vrf.skey"
+      , "--shelley-operational-certificate", "/devnet/opcert.cert"
+      , "--byron-delegation-certificate", "/devnet/byron-delegation.cert"
+      , "--byron-signing-key", "/devnet/byron-delegate.key"
+      ]
+
+  hydra-node-1:
```
[the remark this answers is not in the document]
Comment (nazrhom) [reply docker-compose.yaml:24]: TBH I don't think it is worth having them for such a contained file, they make the document harder to read and reason about (also I never see them used).
Also, most keys have slight differences, so we would really only use them for the volumes and profiles

File: scripts/reset-devnet.sh [hunk 7]
```diff
@@ -4,15 +4,16 @@
 # "up-to-date" genesis files. If the directory exists, it is wiped out.
 set -e
 
-TARGETDIR="devnet"
+ROOTDIR="."
```
Comment (uhbif19) [inline scripts/reset-devnet.sh:7]: Nice.

File: scripts/spin-up-new-devnet.sh [hunk 8]
```diff
@@ -1,10 +1,12 @@
 #!/bin/sh -e
+parent_path=$( cd "$(dirname "${BASH_SOURCE[0]}")" ; pwd -P )
```
Comment (uhbif19) [inline scripts/spin-up-new-devnet.sh:2]: Do not understand why this is required, but okay.

File: stop-demo.sh (hunk 1 above)
Comment (uhbif19) [reply stop-demo.sh]: Does not `depends_on` does this?

While I am a little bit out of sync with latest docker changes, so I may miss something.

**Review by uhbif19 — APPROVED** [review]

**Pull request merged**